### ISSUE TITLE
New Rule: no-trailing-heading-punctuation

### ISSUE THREAD

**Issue by FloEdelmann** [eslint/markdown#682]
### Rule details

Prevent headings that have a punctuation character as the last character in the line

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
Bad:
# This is a heading.

Good:
# This is a heading
```

### Prior Art

Markdownlint rule `MD026` (`no-trailing-punctuation`): https://github.com/DavidAnson/markdownlint/blob/main/doc/md026.md

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by lumirlumir** [comment]
Thank you for opening an issue. However, under our policy, we don’t accept stylistic rules in ESLint core, and existing stylistic rules are feature-frozen.

Please read our policy on this here: https://eslint.org/docs/latest/use/core-concepts/glossary#stylistic-rule

Thank you for understanding.

(FYI, there’s a community plugin that accepts stylistic rules. If you’re interested, you could submit an issue or PR there: https://github.com/lumirlumir/npm-eslint-markdown)
